supervisor/shared/serial.c

```diff
@@ -32,26 +32,72 @@
 #include "shared-bindings/terminalio/Terminal.h"
 #include "supervisor/serial.h"
 #include "supervisor/usb.h"
+#include "shared-bindings/microcontroller/Pin.h"
 
 #include "tusb.h"
 
+#if defined(DEBUG_UART_TX) && defined(DEBUG_UART_RX)
+#include "shared-bindings/busio/UART.h"
+busio_uart_obj_t debug_uart;
+byte buf_array[64];
+ringbuf_t rbuf;
+#endif
+
+void serial_early_init(void) {
+#if defined(DEBUG_UART_TX) && defined(DEBUG_UART_RX)
+    rbuf = (ringbuf_t){ buf_array, sizeof(buf_array) };
+    assert_pin_free(DEBUG_UART_TX);
+    assert_pin_free(DEBUG_UART_RX);
```

Comment:
No need for this. If it fails it'll safe mode because the VM isn't running and it'll try to raise an exception.

Reply:
Removed.